queries.sql

```diff
@@ -23,4 +23,88 @@ SELECT * from animals WHERE name NOT like 'Gabumon';
 
 SELECT *
 FROM animals
-WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+BEGIN;
+UPDATE animals SET weight_kg = 22 WHERE NAME LIKE 'Ditto';
+ROLLBACK;
+COMMIT;
+
+
+BEGIN;
+SAVEPOINT speciesbefore;
+
+UPDATE animals
+SET species = 'unspecified';
+
+ROLLBACK TO speciesbefore;
+
+COMMIT;
+
+
+BEGIN;
+
+UPDATE animals
+SET species = 'digimon' WHERE name LIKE '%mon';
+
+UPDATE animals
+SET species = 'pokemon' WHERE species IS NULL ;
+
+COMMIT;
+
+
+BEGIN;
+
+SAVEPOINT beforecaos;
+
+DELETE FROM animals;
+
+ROLLBACK TO beforecaos;
+
+COMMIT;
+
+
+BEGIN;
+
+SAVEPOINT beforecaos;
+DELETE FROM animals WHERE  date_of_birth > '2022-01-01';
+
+ROLLBACK TO beforecaos;
+COMMIT;
+
+/*
+How many animals are there?
+*/
+SELECT COUNT(*) FROM animals;
+/*
+How many animals have never tried to escape?
+*/
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+/*
+What is the average weight of animals?
+*/
+SELECT SUM(weight_kg) / COUNT(*) AS weight_kg_average
+FROM animals;
+/*
+Who escapes the most, neutered or not neutered animals?
+*/
+SELECT COUNT(escape_attempts) FROM animals WHERE neutered = TRUE;
+SELECT COUNT(escape_attempts) FROM animals WHERE neutered = FALSE;
+/*
+What is the minimum and maximum weight of each type of animal?
+*/
+SELECT weight_kg
+FROM animals
+ORDER BY weight_kg ASC LIMIT 1;
+SELECT weight_kg
+FROM animals
+ORDER BY weight_kg DESC LIMIT 1;
```

Comment:
- [ ] Great job on your query so far! 😄 However, to ensure an accurate result, a small adjustment is needed. Based on the data you've inserted into your database, you should consider selecting the `species` column from the `animal` table instead of `weight_kg` to determine the `max` and `min` weight of each type of animal.

I recommend rewriting your query to achieve the desired output. Here's a revised version:

```sql
SELECT
    species,
    MAX(weight_kg) AS max_weight,
    MIN(weight_kg) AS min_weight
FROM
    animal
GROUP BY
    species;
```

This query will correctly provide you with the maximum and minimum weights for each type of animal in your database.

Happy coding! 😊👍

Reply:
![14](https://github.com/CarlosZ96/first-look-at-sql/assets/125598459/68033590-0ae5-47c0-9d29-d9933123740e)
-Corrected!